### PR TITLE
CNDIT-1181 Call stored proc to populate PublicHealthCaseFact table

### PR DESCRIPTION
**Pull request by sveselev**
## Description ##

Update the Investigation preprocessing service to call the stored procedure `sp_public_health_case_fact_datamart_event` to populate `PublicHealthCaseFact` - a data source for most of the Standard reports in NBS.

Unit tests will be added under context of [CNDIT-1153](https://cdc-nbs.atlassian.net/browse/CNDIT-1153)

## Jira ##

[CNDIT-1181: investigation java service - Update to Call the sp_PublicHealthCaseFact_Datamart for each topic updates](https://cdc-nbs.atlassian.net/browse/CNDIT-1181)

### CODE REVIEW

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java [hunk 1]
```diff
@@ -57,20 +59,24 @@ public String processInvestigation(String value) {
             JsonNode payloadNode = jsonNode.get("payload").path("after");
             if (payloadNode != null && payloadNode.has("public_health_case_uid")) {
                 String publicHealthCaseUid = payloadNode.get("public_health_case_uid").asText();
+
+                // Calling sp_public_health_case_fact_datamart_event
+                logger.info("Executing stored proc with ids: {} to populate PHS fact datamart", publicHealthCaseUid);
```
Comment (rseshadri99) [inline data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java:64]: @sveselev small type PHS should be PHC

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java [hunk 2]
```diff
@@ -57,20 +59,24 @@ public String processInvestigation(String value) {
...
                 if(investigationData.isPresent()) {
                     InvestigationTransformed investigationTransformed = processDataUtil.transformInvestigationData(investigationData.get());
                     kafkaTemplate.send(investigationTopicOutputTransformed, investigationTransformed.toString());
-                    return objectMapper.writeValueAsString(investigationData);
+                    return objectMapper.writeValueAsString(investigationData.get());
                 }
             }
+
         } catch (Exception e) {
             logger.error("Error processing investigation: {}", e.getMessage());
```
Comment (rseshadri99) [inline data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java:78]: @sveselev how feasible is it to also include the list of Ids that are passed with the error message.?

**Review by rseshadri99 — CHANGES_REQUESTED** [review]

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java (hunk 1 above)
Comment (sveselev) [reply data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/service/InvestigationService.java]: @rseshadri99 fixed

**Review by rseshadri99 — APPROVED** [review]
@sveselev looks good

File: data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/repository/InvestigationRepository.java [hunk 3]
```diff
@@ -13,4 +13,7 @@ public interface InvestigationRepository extends JpaRepository<Investigation, St
 
     @Query(nativeQuery = true, value = "execute sp_investigation_event :investigation_uids")
     Optional<Investigation> computeInvestigations(@Param("investigation_uids") String investigationUids);
+
+    @Query(nativeQuery = true, value = "exec sp_public_health_case_fact_datamart_event :investigation_uids")
```
Comment (alog-enquizit) [inline data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/repository/InvestigationRepository.java:17]: We might want to add the stored procs to the liquibase script.

**Pull request merged**